src/libopenrave/kinbody.cpp

```diff
@@ -2098,14 +2107,7 @@ void KinBody::SetDOFValues(const std::vector<dReal>& vJointValues, const Transfo
     if( _veclinks.size() == 0 ) {
         return;
     }
-    Transform baseLinkTransform = bodyTransform * _baseLinkInBodyTransform;
-    Transform tbase = baseLinkTransform*_veclinks.at(0)->GetTransform().inverse();
-    _veclinks.at(0)->SetTransform(baseLinkTransform);
-
-    // apply the relative transformation to all links!! (needed for passive joints)
-    for(size_t i = 1; i < _veclinks.size(); ++i) {
-        _veclinks[i]->SetTransform(tbase*_veclinks[i]->GetTransform());
-    }
+    SetTransform(bodyTransform);
```

Comment:
The code was written to not call `SetTransform` to avoid unnecessarily calling `_UpdateGrabbedBodies` and `_PostprocessChangedParameters`. Please revert this change.

Reply:
If it is ensured that not calling `_UpdateGrabbedBodies` and `_PostprocessChangedParameters` will leave some internal state hanging, then this change can be reverted.

Reply:
If  `_UpdateGrabbedBodies` and `_PostprocesChangedParameters` are absolutely necessary (for example, there will be problems if those are not called), then you can leave this change here.

Reply:
If you have strong opinion on keeping this, I can revert. Otherwise let's go with the method call which I think is cleaner.